### PR TITLE
Translation finished (soappp)

### PR DESCRIPTION
**Pull request by soappp9527**

### CODE REVIEW

**Comment by MichaelChirico** [comment]
Thanks so much! @amy17519 please review when you get a chance :)

Ignore the "changes" in the non-`.po` files. Those are changes in `master` that I just merged into `translation` branch, not sure why it's still showing a `diff`

File: po/R-zh_CN.po [hunk 1]
```diff
@@ -508,47 +508,51 @@ msgid ".SDcols missing at the following indices:"
...
-msgstr ""
+msgstr ".SDcols 为数值，但同时具有 +ve 和 -ve 索引"
 
 msgid ".SDcols is numeric but out of bounds [1,"
-msgstr ""
+msgstr ".SDcols 为数值但超出了 [1,"
 
 msgid "] at:"
-msgstr ""
+msgstr "] 的范围"
```
Comment (amy17519) [inline po/R-zh_CN.po]: keep the `:` ?

File: po/R-zh_CN.po [hunk 2]
```diff
@@ -508,47 +508,51 @@ msgid ".SDcols missing at the following indices:"
...
 
 msgid ".SDcols should be column numbers or names"
-msgstr ""
+msgstr ".SDcols 应为列数或是列名"
 
 msgid "Some items of .SDcols are not column names:"
-msgstr ""
+msgstr ".SDcols 中的部份项目不是列名"
```
Comment (amy17519) [inline po/R-zh_CN.po]: keep the `:` ?

File: po/R-zh_CN.po [hunk 3]
```diff
@@ -508,47 +508,51 @@ msgid ".SDcols missing at the following indices:"
...
 "columns can only be added by name."
-msgstr ""
+msgstr ":= 的 LHS 是列的位置但超出了 [1,ncol] 的范围"
+"新列只能以名称的方式新增"
 
 msgid ""
 "LHS of := isn't column names ('character') or positions ('integer' or "
 "'numeric')"
-msgstr ""
+msgstr ":= 的 LHS 不是列名('字符')或列的位置('整数'或'数值)"
```
Comment (amy17519) [inline po/R-zh_CN.po]: missing a `'` after 数值

File: po/R-zh_CN.po [hunk 4]
```diff
@@ -559,83 +563,103 @@ msgid ""
...
 msgid ""
 "Cannot assign to an under-allocated recursively indexed list -- L[[i]][,:=] "
 "syntax is only valid when i is length 1, but it's length"
 msgstr ""
+"无法指定配置不足的递归索引列表"
+"-- L[[i]][,:=] 语法只有在 i 长度为1时有效，但它的長度"
 
 msgid "Internal error -- item '"
-msgstr ""
+msgstr "内部错误 -- 项目"
```
Comment (amy17519) [inline po/R-zh_CN.po]: missing a `'` at the end

File: po/R-zh_CN.po [hunk 5]
```diff
@@ -559,83 +563,103 @@ msgid ""
...
 
 msgid "' not found in names of list"
-msgstr ""
+msgstr "未能在名称列表中找到"
```
Comment (amy17519) [inline po/R-zh_CN.po]: missing a `'` at the beginning

File: po/R-zh_CN.po [hunk 6]
```diff
@@ -559,83 +563,103 @@ msgid ""
...
+msgstr "侦测到使用 strptime() ，已用 as.POSIXct() 包裹"
+"这么做是为了尽量避免列被指定为 POSIXlt"
+"POSIXlt用40个以上的位元组储存日期(相较于 POSIXct 只用8位元组)"
+"请使用 as.POSIXct() 以避免本警告 (此函数会根据需求在内部调用 strptime())"
 
 msgid ""
 "' is not found in calling scope. Looking in calling scope because this "
 "symbol was prefixed with .. in the j= parameter."
 msgstr ""
+"未能在 calling scope 中找到，请查看 calling scope"
```
Comment (amy17519) [inline po/R-zh_CN.po]: missing a `'` at the beginning. `calling scope` can be translated to `调用范围`？

File: po/R-zh_CN.po [hunk 7]
```diff
@@ -559,83 +563,103 @@ msgid ""
...
 msgid ""
 "Internal error: irows is NULL when making join result at R level. Should no "
 "longer happen now we use CsubsetDT earlier."
-msgstr ""
+msgstr "内部错误 : 在 R 生成连接结果时，irows为 NULL "
+"我们已使用了 CsubsetDT，现在不应该再发生了"
 
 msgid "j (the 2nd argument inside [...]) is a single symbol but column name '"
-msgstr ""
+msgstr "j ( [...] 中的第二顺位参数) 是单个符号而非列名"
```
Comment (amy17519) [inline po/R-zh_CN.po]: this sentence does not exclusive say `非列名`. I think it's fine to say `...是单个符号而列名`

Comment (amy17519) [reply po/R-zh_CN.po]: also, missing `'` at the end

File: po/R-zh_CN.po [hunk 8]
```diff
@@ -559,83 +563,103 @@ msgid ""
...
 
 msgid "' is not found. Perhaps you intended DT[, .."
-msgstr ""
+msgstr "未被找到，也许你打算 DT[, .."
```
Comment (amy17519) [inline po/R-zh_CN.po]: you need a `'` at the beginning

File: po/R-zh_CN.po [hunk 9]
```diff
@@ -559,83 +563,103 @@ msgid ""
...
 
 msgid ""
 "]. This difference to data.frame is deliberate and explained in FAQ 1.1."
 msgstr ""
+"] ，在常见问题 1.1 中有解释dat.table与data.frame的差别"
```
Comment (amy17519) [inline po/R-zh_CN.po]: maybe not translate `FAQ`? I am not sure

File: po/R-zh_CN.po [hunk 10]
```diff
@@ -559,83 +563,103 @@ msgid ""
...
 "variable. Try setnames(DT,'.N','N') first."
-msgstr ""
+msgstr "无法对 '.N' 列进行分组，因为与 data.table 特有的 .N 变量冲突"
+"请先尝试 setnames(DT,'.N','N')"
 
 msgid ""
 "The column '.I' can't be grouped because it conflicts with the special .I "
 "variable. Try setnames(DT,'.I','I') first."
-msgstr ""
+msgstr "无法对 '.l' 列进行分组，因为与 data.table 特有的 .l 变量冲突"
```
Comment (amy17519) [inline po/R-zh_CN.po]: It's `.I` not `.l` here. upper case i

File: po/zh_CN.po [hunk 11]
```diff
@@ -975,13 +975,16 @@ msgid ""
 "All items in j=list(...) should be atomic vectors or lists. If you are "
 "trying something like j=list(.SD,newcol=mean(colA)) then use := by group "
 "instead (much quicker), or cbind or merge afterwards."
-msgstr ""
+msgstr "j=list(...) 中的所有项目必须是原子向量或列表"
+"如果您试图进行 j=list(.SD,newcol=mean(colA)) 之类的的操作"
```
Comment (amy17519) [inline po/zh_CN.po]: extra `的` here

File: po/zh_CN.po [hunk 12]
```diff
@@ -1076,24 +1095,27 @@ msgid ""
...
+msgstr "内部错误 : 区块 0 [%d] 与区块 1 [%d] 都运行了"
 
 #: dogroups.c:456
 #, c-format
 msgid ""
 "\n"
 "  %s took %.3fs for %d groups\n"
-msgstr ""
+msgstr "\n"
+" %s 花了 %s.3fs 在 %d 个组\n"
```
Comment (amy17519) [inline po/zh_CN.po]: `%s.3fs` has an extra `s`. Should be `%.3fs`

File: po/zh_CN.po [hunk 13]
```diff
@@ -1076,24 +1095,27 @@ msgid ""
...
-msgstr ""
+msgstr "提供 %d 项给第 %d 列的第 %d 组，这组有 %d 行"
+"RHS長度必須為 1 (可以使用單個值) 或與 LHS 長度完全匹配"
+"如果您想回收(recycle) RHS，请使用 rep() 向你的代码读者明确表达你的意图"
 
 #: dogroups.c:444
 #, c-format
 msgid "Wrote less rows (%d) than allocated (%d).\n"
-msgstr ""
+msgstr "写入的行 (%d) 少于分配的 (%d)\n"
```
Comment (amy17519) [inline po/zh_CN.po]: nbd but missing a period

File: po/zh_CN.po [hunk 14]
```diff
@@ -1076,24 +1095,27 @@ msgid ""
 "length must either be 1 (single values are ok) or match the LHS length "
 "exactly. If you wish to 'recycle' the RHS please use rep() explicitly to "
 "make this intent clear to readers of your code."
-msgstr ""
+msgstr "提供 %d 项给第 %d 列的第 %d 组，这组有 %d 行"
```
Comment (amy17519) [inline po/zh_CN.po]: should be `提供 %d 项给第 %d 组的第 %d 列，这组有 %d 行`

**Comment by amy17519** [comment]
added some comments! Thanks for working on this! @soappp9527

File: po/R-zh_CN.po (hunk 9 above)
Comment (MichaelChirico) [reply po/R-zh_CN.po]: agree, FAQ should be kept here

File: po/zh_CN.po (hunk 14 above)
Comment (soappp9527) [reply po/zh_CN.po]: but I think the the order of `%d` should be same as original sentence, right? if I exchange the `group` and `column` position won't the number fill in `%d` mismatch?

File: po/zh_CN.po (hunk 13 above)
Comment (soappp9527) [reply po/zh_CN.po]: Thanks for your comment, but I don`t really understand what you mean.

File: po/zh_CN.po (hunk 11 above)
Comment (soappp9527) [reply po/zh_CN.po]: Thanks for your comment, but I don`t really understand what you mean.

Comment (MichaelChirico) [reply po/zh_CN.po]: in the second line, 的 is doubled, is that intentional?

File: po/zh_CN.po (hunk 13 above)
Comment (MichaelChirico) [reply po/zh_CN.po]: nbd means "no big deal". after `(%d)` you're missing `.`

File: po/zh_CN.po (hunk 14 above)
Comment (MichaelChirico) [reply po/zh_CN.po]: there's a bit in the FAQ about this (sorry, can't link on mobile)

You should write it in the order that's most natural, _but_ if you don't respect the original order, you need to use references -- the first %d (in the original message) should be marked as %1$d (and so on for the other messages)

File: po/zh_CN.po (hunk 11 above)
Comment (soappp9527) [reply po/zh_CN.po]: oh, I see that. Thank you! mixing Chinese and English makes me confused.

File: po/zh_CN.po (hunk 14 above)
Comment (soappp9527) [reply po/zh_CN.po]: got that

File: po/zh_CN.po (hunk 13 above)
Comment (soappp9527) [reply po/zh_CN.po]: got that. I forgot period also mean `.`.

**Comment by MichaelChirico** [comment]
Looks all set! Thanks @soappp9527  and @amy17519 !!!

**Pull request merged**